# Data Dict Analysis 

A notebook for analysing two databases and exploring it in 2024_06

É possível encontrar os dados em

https://dados.ons.org.br/dataset/restricao_coff_eolica_usi

https://dados.ons.org.br/dataset/restricao_coff_eolica_detail

In [1]:
import pandas as pd

In [2]:
df_constrainedoff = pd.read_csv('../data_raw/RESTRICAO_COFF_EOLICA_2024_06.csv', delimiter=';')
df_constrainedoff_by_plant= pd.read_csv('../data_raw/RESTRICAO_COFF_EOLICA_DETAIL_2024_06.csv', delimiter=';')

In [3]:
display(df_constrainedoff)
display(df_constrainedoff_by_plant)

id_subsistema nom_subsistema id_estado      nom_estado  \
0                  N          NORTE        MA        MARANHAO   
1                  N          NORTE        MA        MARANHAO   
2                  N          NORTE        MA        MARANHAO   
3                  N          NORTE        MA        MARANHAO   
4                  N          NORTE        MA        MARANHAO   
...              ...            ...       ...             ...   
228955             S            SUL        SC  SANTA CATARINA   
228956             S            SUL        SC  SANTA CATARINA   
228957             S            SUL        SC  SANTA CATARINA   
228958             S            SUL        SC  SANTA CATARINA   
228959             S            SUL        SC  SANTA CATARINA   

                  nom_usina      id_ons ceg         din_instante  val_geracao  \
0       CONJ. PAULINO NEVES   CJU_MAPLN   -  2024-06-01 00:00:00        0.000   
1       CONJ. PAULINO NEVES   CJU_MAPLN   -  2024-06-01 00:30:00        2.925   
2       CONJ. PAULINO NEVES   CJU_MAPLN   -  2024-06-01 01:00:00       13.215   
3       CONJ. PAULINO NEVES   CJU_MAPLN   -  2024-06-01 01:30:00        5.892   
4       CONJ. PAULINO NEVES   CJU_MAPLN   -  2024-06-01 02:00:00       17.770   
...                     ...         ...  ..                  ...          ...   
228955     CONJ. BOM JARDIM  CJU_SCCEBJ   -  2024-06-30 21:30:00       23.722   
228956     CONJ. BOM JARDIM  CJU_SCCEBJ   -  2024-06-30 22:00:00       24.517   
228957     CONJ. BOM JARDIM  CJU_SCCEBJ   -  2024-06-30 22:30:00       27.724   
228958     CONJ. BOM JARDIM  CJU_SCCEBJ   -  2024-06-30 23:00:00       32.936   
228959     CONJ. BOM JARDIM  CJU_SCCEBJ   -  2024-06-30 23:30:00       29.526   

        val_geracaolimitada  val_disponibilidade  val_geracaoreferencia  \
0                       NaN              118.360                 11.236   
1                       NaN              191.261                 16.769   
2                       NaN              324.763                 30.001   
3                       NaN              301.497                 23.664   
4                       NaN              319.832                 33.905   
...                     ...                  ...                    ...   
228955                  NaN               87.000                 21.411   
228956                  NaN               87.000                 19.014   
228957                  NaN               87.000                 28.744   
228958                  NaN               87.000                 37.228   
228959                  NaN               87.000                 32.573   

        val_geracaoreferenciafinal cod_razaorestricao cod_origemrestricao  
0                              NaN                NaN                 NaN  
1                              NaN                NaN                 NaN  
2                              NaN                NaN                 NaN  
3                              NaN                NaN                 NaN  
4                              NaN                NaN                 NaN  
...                            ...                ...                 ...  
228955                         NaN                NaN                 NaN  
228956                         NaN                NaN                 NaN  
228957                         NaN                NaN                 NaN  
228958                         NaN                NaN                 NaN  
228959                         NaN                NaN                 NaN  

[228960 rows x 15 columns]

id_subsistema id_estado nom_modalidadeoperacao    nom_conjuntousina  \
0                   N        MA              Tipo II-C  Conj. Paulino Neves   
1                   N        MA              Tipo II-C  Conj. Paulino Neves   
2                   N        MA              Tipo II-C  Conj. Paulino Neves   
3                   N        MA              Tipo II-C  Conj. Paulino Neves   
4                   N        MA              Tipo II-C  Conj. Paulino Neves   
...               ...       ...                    ...                  ...   
1398283             S        SC              Tipo II-C      Conj. Água Doce   
1398284             S        SC              Tipo II-C     Conj. Bom Jardim   
1398285             S        SC              Tipo II-C     Conj. Bom Jardim   
1398286             S        SC              Tipo II-C      Conj. Água Doce   
1398287             S        SC              Tipo II-C     Conj. Bom Jardim   

             nom_usina  id_ons                    ceg         din_instante  \
0            Delta 3 I  MAEDT1  EOL.CV.MA.033682-3.01  2024-06-01 00:00:00   
1           Delta 3 II  MAEDT2  EOL.CV.MA.033683-1.01  2024-06-01 00:00:00   
2          Delta 3 III  MAEDT3  EOL.CV.MA.033684-0.01  2024-06-01 00:00:00   
3           Delta 3 IV  MAEDT4  EOL.CV.MA.033685-8.01  2024-06-01 00:00:00   
4            Delta 3 V  MAEDT5  EOL.CV.MA.033675-0.01  2024-06-01 00:00:00   
...                ...     ...                    ...                  ...   
1398283      Cruz Alta  SCECAL  EOL.CV.SC.029070-0.01  2024-06-30 23:30:00   
1398284        Púlpito  SCUPPT  EOL.CV.SC.029051-3.01  2024-06-30 23:30:00   
1398285    Rio do Ouro  SCUROU  EOL.CV.SC.029055-6.01  2024-06-30 23:30:00   
1398286          Salto  SCESTO  EOL.CV.SC.029056-4.01  2024-06-30 23:30:00   
1398287  Santo Antônio  SCUSTO  EOL.CV.SC.029053-0.01  2024-06-30 23:30:00   

         val_ventoverificado  flg_dadoventoinvalido  val_geracaoestimada  \
0                      2.038                    0.0                0.383   
1                      2.680                    0.0                0.801   
2                      2.670                    0.0                0.560   
3                      2.418                    0.0                1.379   
4                      2.567                    0.0                1.443   
...                      ...                    ...                  ...   
1398283                3.715                    1.0                0.000   
1398284                7.382                    0.0               11.008   
1398285                7.378                    0.0                9.796   
1398286                3.715                    1.0                0.000   
1398287                7.382                    0.0                0.882   

         val_geracaoverificada  
0                        0.000  
1                        0.000  
2                        0.000  
3                        0.000  
4                        0.000  
...                        ...  
1398283                  0.023  
1398284                  9.334  
1398285                  8.920  
1398286                  0.006  
1398287                  1.184  

[1398288 rows x 12 columns]

In [4]:
# Converter 'nom_usina' para maiúsculas em ambas as bases
df_constrainedoff['nom_usina'] = df_constrainedoff['nom_usina'].str.upper()
df_constrainedoff_by_plant['nom_conjuntousina'] = df_constrainedoff_by_plant['nom_conjuntousina'].str.upper()

df_constrainedoff_by_plant.head()

id_subsistema id_estado nom_modalidadeoperacao    nom_conjuntousina  \
0             N        MA              Tipo II-C  CONJ. PAULINO NEVES   
1             N        MA              Tipo II-C  CONJ. PAULINO NEVES   
2             N        MA              Tipo II-C  CONJ. PAULINO NEVES   
3             N        MA              Tipo II-C  CONJ. PAULINO NEVES   
4             N        MA              Tipo II-C  CONJ. PAULINO NEVES   

     nom_usina  id_ons                    ceg         din_instante  \
0    Delta 3 I  MAEDT1  EOL.CV.MA.033682-3.01  2024-06-01 00:00:00   
1   Delta 3 II  MAEDT2  EOL.CV.MA.033683-1.01  2024-06-01 00:00:00   
2  Delta 3 III  MAEDT3  EOL.CV.MA.033684-0.01  2024-06-01 00:00:00   
3   Delta 3 IV  MAEDT4  EOL.CV.MA.033685-8.01  2024-06-01 00:00:00   
4    Delta 3 V  MAEDT5  EOL.CV.MA.033675-0.01  2024-06-01 00:00:00   

   val_ventoverificado  flg_dadoventoinvalido  val_geracaoestimada  \
0                2.038                    0.0                0.383   
1                2.680                    0.0                0.801   
2                2.670                    0.0                0.560   
3                2.418                    0.0                1.379   
4                2.567                    0.0                1.443   

   val_geracaoverificada  
0                    0.0  
1                    0.0  
2                    0.0  
3                    0.0  
4                    0.0

In [28]:
df_constrainedoff.info()
df_constrainedoff.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 228960 entries, 0 to 228959
Data columns (total 15 columns):
 #   Column                      Non-Null Count   Dtype  
---  ------                      --------------   -----  
 0   id_subsistema               228960 non-null  object 
 1   nom_subsistema              228960 non-null  object 
 2   id_estado                   228960 non-null  object 
 3   nom_estado                  228960 non-null  object 
 4   nom_usina                   228960 non-null  object 
 5   id_ons                      228960 non-null  object 
 6   ceg                         228960 non-null  object 
 7   din_instante                228960 non-null  object 
 8   val_geracao                 228960 non-null  float64
 9   val_geracaolimitada         53673 non-null   float64
 10  val_disponibilidade         228960 non-null  float64
 11  val_geracaoreferencia       228960 non-null  float64
 12  val_geracaoreferenciafinal  2522 non-null    float64
 13  cod_razaorestr

val_geracao  val_geracaolimitada  val_disponibilidade  \
count  228960.000000         53673.000000        228960.000000   
mean       82.132092            68.161052           153.456456   
std        94.381027            86.359749           130.492057   
min         0.000000             0.000000             0.000000   
25%        19.793750             8.900000            63.900000   
50%        48.841500            38.000000           111.047000   
75%       110.349000            92.433000           207.013250   
max       687.337000           671.200000           666.999000   

       val_geracaoreferencia  val_geracaoreferenciafinal  
count          228960.000000                 2522.000000  
mean               75.145526                   48.884427  
std                84.341511                   73.735810  
min                 0.000000                    0.000000  
25%                17.737750                    0.000000  
50%                44.779000                   14.819000  
75%               102.286000                   70.381250  
max               577.399000                  471.440000

In [29]:
df_constrainedoff_by_plant.info()
df_constrainedoff_by_plant.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1398288 entries, 0 to 1398287
Data columns (total 12 columns):
 #   Column                  Non-Null Count    Dtype  
---  ------                  --------------    -----  
 0   id_subsistema           1398288 non-null  object 
 1   id_estado               1398288 non-null  object 
 2   nom_modalidadeoperacao  1398288 non-null  object 
 3   nom_conjuntousina       1382448 non-null  object 
 4   nom_usina               1398288 non-null  object 
 5   id_ons                  1398288 non-null  object 
 6   ceg                     1398288 non-null  object 
 7   din_instante            1398288 non-null  object 
 8   val_ventoverificado     1398288 non-null  float64
 9   flg_dadoventoinvalido   1398288 non-null  float64
 10  val_geracaoestimada     1398288 non-null  float64
 11  val_geracaoverificada   1398288 non-null  float64
dtypes: float64(4), object(8)
memory usage: 128.0+ MB


val_ventoverificado  flg_dadoventoinvalido  val_geracaoestimada  \
count         1.398288e+06           1.398288e+06         1.398288e+06   
mean          8.072121e+00           2.142792e-01         1.230456e+01   
std           8.355202e+00           4.103215e-01         1.185488e+01   
min          -1.000000e+00           0.000000e+00         0.000000e+00   
25%           6.000000e+00           0.000000e+00         1.734000e+00   
50%           7.812000e+00           0.000000e+00         9.535000e+00   
75%           9.713000e+00           0.000000e+00         1.980500e+01   
max           2.410000e+02           1.000000e+00         7.735000e+01   

       val_geracaoverificada  
count           1.398288e+06  
mean            1.334290e+01  
std             1.161522e+01  
min             0.000000e+00  
25%             4.431000e+00  
50%             1.070800e+01  
75%             2.017900e+01  
max             2.770000e+02

# MA Data
Vamos analisar apenas os padrões de Paulino Neves

In [5]:
df_constrainedoff_ma = df_constrainedoff[df_constrainedoff['id_estado'] == 'MA']
df_constrainedoff_by_plant_ma = df_constrainedoff_by_plant[df_constrainedoff_by_plant['id_estado'] == 'MA']

# display(df_constrainedoff_ma)
# display(df_constrainedoff_by_plant_ma)

Na análise abaixo, qualquer dado que é analisado limitando algum Delta, o valor das linhas é de 1440, que é igual ao valor de linhas em df_constrainedoff_ma

21600/1440 = 15 usinas

Ao utilizar unique para listar todas as nom_usinas, tivemos o total de 15 usinas listadas
['Delta 3 I' 'Delta 3 II' 'Delta 3 III' 'Delta 3 IV' 'Delta 3 V'
 'Delta 3 VI' 'Delta 3 VII' 'Delta 3 VIII' 'Delta 5 I' 'Delta 5 II'
 'Delta 6 I' 'Delta 6 II' 'Delta 7 I' 'Delta 7 II' 'Delta 8 I']

 Minha suposição é que nom_conjuntousina é o correspondente ao nome do cluster, já nom_usina são correspondentes aos nomes de registro/parque que compõem o cluster como um todo

In [6]:
df_delta = df_constrainedoff_by_plant[(df_constrainedoff_by_plant['nom_conjuntousina'] == 'CONJ. PAULINO NEVES') & (df_constrainedoff_by_plant['nom_usina'] == 'Delta 8 I')]

display(df_delta)

id_subsistema id_estado nom_modalidadeoperacao    nom_conjuntousina  \
14                  N        MA              Tipo II-C  CONJ. PAULINO NEVES   
979                 N        MA              Tipo II-C  CONJ. PAULINO NEVES   
1944                N        MA              Tipo II-C  CONJ. PAULINO NEVES   
2909                N        MA              Tipo II-C  CONJ. PAULINO NEVES   
3874                N        MA              Tipo II-C  CONJ. PAULINO NEVES   
...               ...       ...                    ...                  ...   
1393432             N        MA              Tipo II-C  CONJ. PAULINO NEVES   
1394406             N        MA              Tipo II-C  CONJ. PAULINO NEVES   
1395380             N        MA              Tipo II-C  CONJ. PAULINO NEVES   
1396354             N        MA              Tipo II-C  CONJ. PAULINO NEVES   
1397328             N        MA              Tipo II-C  CONJ. PAULINO NEVES   

         nom_usina  id_ons                    ceg         din_instante  \
14       Delta 8 I  MAED8I  EOL.CV.MA.040574-4.01  2024-06-01 00:00:00   
979      Delta 8 I  MAED8I  EOL.CV.MA.040574-4.01  2024-06-01 00:30:00   
1944     Delta 8 I  MAED8I  EOL.CV.MA.040574-4.01  2024-06-01 01:00:00   
2909     Delta 8 I  MAED8I  EOL.CV.MA.040574-4.01  2024-06-01 01:30:00   
3874     Delta 8 I  MAED8I  EOL.CV.MA.040574-4.01  2024-06-01 02:00:00   
...            ...     ...                    ...                  ...   
1393432  Delta 8 I  MAED8I  EOL.CV.MA.040574-4.01  2024-06-30 21:30:00   
1394406  Delta 8 I  MAED8I  EOL.CV.MA.040574-4.01  2024-06-30 22:00:00   
1395380  Delta 8 I  MAED8I  EOL.CV.MA.040574-4.01  2024-06-30 22:30:00   
1396354  Delta 8 I  MAED8I  EOL.CV.MA.040574-4.01  2024-06-30 23:00:00   
1397328  Delta 8 I  MAED8I  EOL.CV.MA.040574-4.01  2024-06-30 23:30:00   

         val_ventoverificado  flg_dadoventoinvalido  val_geracaoestimada  \
14                     2.185                    0.0                0.695   
979                    2.996                    0.0                1.344   
1944                   4.180                    0.0                3.013   
2909                   3.787                    0.0                2.338   
3874                   4.438                    0.0                3.536   
...                      ...                    ...                  ...   
1393432                7.312                    0.0               14.781   
1394406                7.621                    0.0               16.486   
1395380                7.709                    0.0               16.976   
1396354                7.555                    0.0               16.121   
1397328                7.483                    0.0               15.721   

         val_geracaoverificada  
14                       0.000  
979                      0.000  
1944                     0.067  
2909                     0.000  
3874                     0.591  
...                        ...  
1393432                 11.141  
1394406                 10.191  
1395380                  9.742  
1396354                  9.088  
1397328                  9.250  

[1440 rows x 12 columns]

In [7]:
nom_usinas_unicas = df_constrainedoff_by_plant_ma['nom_usina'].unique()
print(nom_usinas_unicas)

['Delta 3 I' 'Delta 3 II' 'Delta 3 III' 'Delta 3 IV' 'Delta 3 V'
 'Delta 3 VI' 'Delta 3 VII' 'Delta 3 VIII' 'Delta 5 I' 'Delta 5 II'
 'Delta 6 I' 'Delta 6 II' 'Delta 7 I' 'Delta 7 II' 'Delta 8 I']


# Limitation Explore
- 53673 linhas de código em que há clusters que sofreram algum tipo de restrição
- As restrições são classificadas como ENE, REL ou CNF
- Quando há restrição documentada, não possuem lacunas em relação a 'val_geracaolimitada'

In [12]:
# Verificar onde o cod_restricao não é NULL
data_with_restrition = df_constrainedoff[df_constrainedoff['cod_razaorestricao'].notnull()]
data_with_restrition

id_subsistema nom_subsistema id_estado      nom_estado  \
14                 N          NORTE        MA        MARANHAO   
15                 N          NORTE        MA        MARANHAO   
16                 N          NORTE        MA        MARANHAO   
17                 N          NORTE        MA        MARANHAO   
18                 N          NORTE        MA        MARANHAO   
...              ...            ...       ...             ...   
228937             S            SUL        SC  SANTA CATARINA   
228938             S            SUL        SC  SANTA CATARINA   
228939             S            SUL        SC  SANTA CATARINA   
228940             S            SUL        SC  SANTA CATARINA   
228941             S            SUL        SC  SANTA CATARINA   

                  nom_usina      id_ons ceg         din_instante  val_geracao  \
14      CONJ. PAULINO NEVES   CJU_MAPLN   -  2024-06-01 07:00:00       63.053   
15      CONJ. PAULINO NEVES   CJU_MAPLN   -  2024-06-01 07:30:00       63.935   
16      CONJ. PAULINO NEVES   CJU_MAPLN   -  2024-06-01 08:00:00       55.850   
17      CONJ. PAULINO NEVES   CJU_MAPLN   -  2024-06-01 08:30:00       49.878   
18      CONJ. PAULINO NEVES   CJU_MAPLN   -  2024-06-01 09:00:00       46.990   
...                     ...         ...  ..                  ...          ...   
228937     CONJ. BOM JARDIM  CJU_SCCEBJ   -  2024-06-30 12:30:00       50.611   
228938     CONJ. BOM JARDIM  CJU_SCCEBJ   -  2024-06-30 13:00:00       49.010   
228939     CONJ. BOM JARDIM  CJU_SCCEBJ   -  2024-06-30 13:30:00       45.911   
228940     CONJ. BOM JARDIM  CJU_SCCEBJ   -  2024-06-30 14:00:00       34.394   
228941     CONJ. BOM JARDIM  CJU_SCCEBJ   -  2024-06-30 14:30:00       30.128   

        val_geracaolimitada  val_disponibilidade  val_geracaoreferencia  \
14                   85.396              369.759                123.490   
15                   59.700              370.050                129.143   
16                   55.333              372.500                250.838   
17                   54.933              373.981                336.761   
18                   53.933              374.137                325.520   
...                     ...                  ...                    ...   
228937               53.000               88.500                 78.800   
228938               53.000               88.500                 64.973   
228939               53.000               88.500                 56.638   
228940               53.000               88.500                 33.404   
228941               43.723               88.500                 27.699   

        val_geracaoreferenciafinal cod_razaorestricao cod_origemrestricao  
14                             NaN                ENE                 SIS  
15                             NaN                ENE                 SIS  
16                             NaN                ENE                 SIS  
17                             NaN                ENE                 SIS  
18                             NaN                ENE                 SIS  
...                            ...                ...                 ...  
228937                         NaN                ENE                 SIS  
228938                         NaN                ENE                 SIS  
228939                         NaN                ENE                 SIS  
228940                         NaN                ENE                 SIS  
228941                         NaN                ENE                 SIS  

[53673 rows x 15 columns]

In [13]:
data_with_restrition['cod_razaorestricao'].unique()

array(['ENE', 'REL', 'CNF'], dtype=object)

In [19]:
# Verificar quantas linhas possuem nom_usina = 'CONJ. PAULINO NEVES'
data_with_restrition[data_with_restrition['nom_usina'] == 'CONJ. PAULINO NEVES']

id_subsistema nom_subsistema id_estado nom_estado            nom_usina  \
14               N          NORTE        MA   MARANHAO  CONJ. PAULINO NEVES   
15               N          NORTE        MA   MARANHAO  CONJ. PAULINO NEVES   
16               N          NORTE        MA   MARANHAO  CONJ. PAULINO NEVES   
17               N          NORTE        MA   MARANHAO  CONJ. PAULINO NEVES   
18               N          NORTE        MA   MARANHAO  CONJ. PAULINO NEVES   
...            ...            ...       ...        ...                  ...   
1417             N          NORTE        MA   MARANHAO  CONJ. PAULINO NEVES   
1418             N          NORTE        MA   MARANHAO  CONJ. PAULINO NEVES   
1419             N          NORTE        MA   MARANHAO  CONJ. PAULINO NEVES   
1420             N          NORTE        MA   MARANHAO  CONJ. PAULINO NEVES   
1421             N          NORTE        MA   MARANHAO  CONJ. PAULINO NEVES   

         id_ons ceg         din_instante  val_geracao  val_geracaolimitada  \
14    CJU_MAPLN   -  2024-06-01 07:00:00       63.053               85.396   
15    CJU_MAPLN   -  2024-06-01 07:30:00       63.935               59.700   
16    CJU_MAPLN   -  2024-06-01 08:00:00       55.850               55.333   
17    CJU_MAPLN   -  2024-06-01 08:30:00       49.878               54.933   
18    CJU_MAPLN   -  2024-06-01 09:00:00       46.990               53.933   
...         ...  ..                  ...          ...                  ...   
1417  CJU_MAPLN   -  2024-06-30 12:30:00       31.774               41.000   
1418  CJU_MAPLN   -  2024-06-30 13:00:00       31.742               41.000   
1419  CJU_MAPLN   -  2024-06-30 13:30:00       32.293               41.000   
1420  CJU_MAPLN   -  2024-06-30 14:00:00       54.617               68.000   
1421  CJU_MAPLN   -  2024-06-30 14:30:00      118.465              158.847   

      val_disponibilidade  val_geracaoreferencia  val_geracaoreferenciafinal  \
14                369.759                123.490                         NaN   
15                370.050                129.143                         NaN   
16                372.500                250.838                         NaN   
17                373.981                336.761                         NaN   
18                374.137                325.520                         NaN   
...                   ...                    ...                         ...   
1417              332.427                369.907                         NaN   
1418              332.760                366.804                         NaN   
1419              332.299                346.869                         NaN   
1420              346.190                325.784                         NaN   
1421              356.566                310.582                         NaN   

     cod_razaorestricao cod_origemrestricao  
14                  ENE                 SIS  
15                  ENE                 SIS  
16                  ENE                 SIS  
17                  ENE                 SIS  
18                  ENE                 SIS  
...                 ...                 ...  
1417                ENE                 SIS  
1418                ENE                 SIS  
1419                ENE                 SIS  
1420                ENE                 SIS  
1421                ENE                 SIS  

[97 rows x 15 columns]

In [21]:
# Verificar quantas linhas possuem val_geracaolimitada = null
len(data_with_restrition[data_with_restrition['val_geracaolimitada'].isnull()])

0

In [25]:
len(data_with_restrition[data_with_restrition['val_geracaoreferenciafinal'].notnull()])

2522

In [26]:
data_with_restrition[data_with_restrition['val_geracaoreferenciafinal'].notnull()]

id_subsistema nom_subsistema id_estado           nom_estado  \
1478              NE       NORDESTE        BA                BAHIA   
1479              NE       NORDESTE        BA                BAHIA   
1480              NE       NORDESTE        BA                BAHIA   
1481              NE       NORDESTE        BA                BAHIA   
1482              NE       NORDESTE        BA                BAHIA   
...              ...            ...       ...                  ...   
211220            NE       NORDESTE        RN  RIO GRANDE DO NORTE   
211221            NE       NORDESTE        RN  RIO GRANDE DO NORTE   
211222            NE       NORDESTE        RN  RIO GRANDE DO NORTE   
211223            NE       NORDESTE        RN  RIO GRANDE DO NORTE   
211224            NE       NORDESTE        RN  RIO GRANDE DO NORTE   

               nom_usina     id_ons                    ceg  \
1478        CONJ. ABIL I  CJU_BAABL                      -   
1479        CONJ. ABIL I  CJU_BAABL                      -   
1480        CONJ. ABIL I  CJU_BAABL                      -   
1481        CONJ. ABIL I  CJU_BAABL                      -   
1482        CONJ. ABIL I  CJU_BAABL                      -   
...                  ...        ...                    ...   
211220  REI DOS VENTOS 3     RNUERT  EOL.CV.RN.030417-4.01   
211221  REI DOS VENTOS 3     RNUERT  EOL.CV.RN.030417-4.01   
211222  REI DOS VENTOS 3     RNUERT  EOL.CV.RN.030417-4.01   
211223  REI DOS VENTOS 3     RNUERT  EOL.CV.RN.030417-4.01   
211224  REI DOS VENTOS 3     RNUERT  EOL.CV.RN.030417-4.01   

               din_instante  val_geracao  val_geracaolimitada  \
1478    2024-06-01 19:00:00       21.374               26.723   
1479    2024-06-01 19:30:00        0.275                0.000   
1480    2024-06-01 20:00:00        0.000                0.000   
1481    2024-06-01 20:30:00        0.201                0.233   
1482    2024-06-01 21:00:00       14.208               24.233   
...                     ...          ...                  ...   
211220  2024-06-21 10:00:00        8.949               11.000   
211221  2024-06-21 10:30:00        8.753               11.000   
211222  2024-06-21 11:00:00        8.661               11.000   
211223  2024-06-21 11:30:00        8.579               11.000   
211224  2024-06-21 12:00:00        7.270                8.600   

        val_disponibilidade  val_geracaoreferencia  \
1478                 68.100                 61.667   
1479                 51.198                 66.335   
1480                 50.700                 45.400   
1481                 50.700                 66.756   
1482                 50.700                 64.151   
...                     ...                    ...   
211220               52.173                  0.000   
211221               51.437                  0.000   
211222               51.993                  0.000   
211223               51.400                  0.000   
211224               50.321                  0.000   

        val_geracaoreferenciafinal cod_razaorestricao cod_origemrestricao  
1478                         0.000                REL                 SIS  
1479                        51.198                REL                 SIS  
1480                        45.400                REL                 SIS  
1481                         0.000                REL                 SIS  
1482                         0.000                REL                 SIS  
...                            ...                ...                 ...  
211220                       0.000                REL                 SIS  
211221                       0.000                REL                 SIS  
211222                       0.000                REL                 SIS  
211223                       0.000                REL                 SIS  
211224                       0.000                REL                 SIS  

[2522 rows x 15 columns]